### PR TITLE
[bsp_lpc55sxx] i2c,rtc bsp update

### PR DESCRIPTION
**Pull request by yandld**
1. add i2c DMA mode
2. enable rtc driver

## 拉取/合并请求描述：(PR description)

[

LPC55S69_EVK

#### 为什么提交这份PR (why to submit this PR)
NXP LPC55Sxx BSP support

#### 在什么测试环境下测试通过 (what is the test environment)
LPC55S69_EVK
]

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 代码是高质量的 Code in this PR is of high quality
- [x] 已经使用[formatting](https://github.com/mysterywolf/formatting)等源码格式化工具确保格式符合[RT-Thread代码规范](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_en.md)

### CODE REVIEW

File: bsp/lpc55sxx/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+lpc55s69_nxp_evk_internal
+Libraries/drivers/formatting.py
```
Comment (mysterywolf) [inline bsp/lpc55sxx/.gitignore:2]: 请确认这两个忽略是否正确，没有这两个文件/文件夹

**Comment by mysterywolf** [comment]
麻烦用formatting脚本把driver扫一下

**Comment by yandld** [comment]
> 麻烦用formatting脚本把driver扫一下

sorry, 已经formmating 修改

File: bsp/lpc55sxx/.gitignore (hunk 1 above)
Comment (yandld) [reply bsp/lpc55sxx/.gitignore]: 确认忽略正确

Comment (mysterywolf) [reply bsp/lpc55sxx/.gitignore]: 路径下没有foramtting.py这个文件 这个文件也不应该上传到仓库上来

**Pull request merged**